_posts/2018-03-19-making-a-components-pod.markdown

```diff
@@ -0,0 +1,486 @@
+react_path = File.join(node_modules_path, 'react-native')
+yoga_path = File.join(react_path, 'ReactCommon/yoga')
+folly_spec_path = File.join(react_path, 'third-party-podspecs/Folly.podspec')
+glog_spec_path = File.join(react_path, 'third-party-podspecs/glog.podspec')
+double_conversion_spec_path = File.join(react_path, 'third-party-podspecs/DoubleConversion.podspec')
+
+plugin 'cocoapods-fix-react-native'
+
+target 'GitDawg_Example' do
+  pod 'GitDawg', path: '../'
+
+  # We want extra developer support in React inside this app
+  pod 'React', path: react_path, subspecs: ['DevSupport']
+
+  # We're letting CP know where it can find these Podspecs
+  pod 'yoga', path: yoga_path
+  pod 'Folly', podspec: folly_spec_path
+  pod 'DoubleConversion', podspec: double_conversion_spec_path
+  pod 'glog', podspec: glog_spec_path
+end
+```
+
+Then run `bundle exec pod install` in the Example dir, which will bring all the React Native deps into your project.
+
+We need some native code to represent our Welcome component from the React Native template. So create two new files in
+`Pod/Classes`:
+
+```sh
+$ touch ../Pod/Classes/GDWelcomeViewController.h ../Pod/Classes/GDWelcomeViewController.m
+```
+
+It is a pretty vanilla `UIViewController`, so declare it exists in the interface and then use an `RCTRootView` as it's
+`self.view`.
+
+```objc
+#import <UIKit/UIKit.h>
+
+@interface GDWelcomeViewController : UIViewController
+@end
+```
+
+We're going to handle the React bridging in this `UIViewController`, because that is the simplest option for our Hello
+World app. We'll be going back to this later.
+
+```objc
+#import "GDWelcomeViewController.h"
+#import <React/RCTRootView.h>
+#import <React/RCTBridgeDelegate.h>
+#import <React/RCTBridge.h>
+
+// Let this View Controller handle getting the URL for the JS
+@interface GDWelcomeViewController () <RCTBridgeDelegate>
+@end
+
+@implementation GDWelcomeViewController
+
+// Set self.view on the VC to be an RCTRootView
+- (void)loadView
+{
+    RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:@{}];
+    self.view = [[RCTRootView alloc] initWithBridge:bridge
+                                         moduleName:@"GitDawg"
+                                  initialProperties:@{}];
+}
+
+// Use our bundled JS for now
+- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
+{
+    NSBundle *gitdawgBundle = [NSBundle bundleForClass:GDWelcomeViewController.class];
+    return [gitdawgBundle URLForResource:@"GitDawg" withExtension:@"js"];
+}
+
+@end
+```
+
+Run `bundle exec pod install` inside the `Example` folder again to update the Pod with the new files. Then you're good
+to go. As the `pod lib create` template uses storyboards, you will need to open up the example app's storyboard and
+change the initial view controller to be a `GDWelcomeViewController`. Then run the app in the simulator, and you should
+get this screen:
+
+<center><img src="/images/making_cp_pod/success.png" width="50%" /></center>
```

Comment:
I wasn't able to get this running, Xcode is crashing on `dispatch_queue_t queue = m_moduleData.methodQueue` in `RCTNativeModule.mm` with the following logs:

```
2018-04-05 17:10:46.817 [info][tid:main][RCTCxxBridge.mm:212] Initializing <RCTCxxBridge: 0x6080001d5ea0> (parent: <RCTBridge: 0x6080000dc7e0>, executor: (null))
2018-04-05 17:10:46.901 [warn][tid:main][RCTBridge.m:120] Class RCTCxxModule was not exported. Did you forget to use RCT_EXPORT_MODULE()?
2018-04-05 17:10:46.917 [warn][tid:main][RCTModuleData.mm:69] Module RCTImageLoader requires main queue setup since it overrides `init` but doesn't implement `requiresMainQueueSetup`. In a future release React Native will default to initializing all native modules on a background thread unless explicitly opted-out of.
2018-04-05 17:10:46.928 [info][tid:main][RCTRootView.m:295] Running application GitDawg ({
    initialProps =     {
    };
    rootTag = 1;
})
```

It appears related to https://github.com/orta/cocoapods-fix-react-native/pull/3

Reply:
This has been resolved.